components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useState } from "react";
+import { Button } from "../../components/Button";
+import { downloadUsageCSV } from "./download-usage-csv";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const [isSaving, setIsSaving] = useState(false);
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        setIsSaving(true);
+
+        try {
+            const downloaded = await downloadUsageCSV({
+                orgName: org?.slug ?? org?.id,
+                attributionId: AttributionId.render(attributionId),
+                from: startDate.startOf("day").valueOf(),
+                to: endDate.endOf("day").valueOf(),
+            });
+
+            if (!downloaded) {
+                toast("There are no usage records for that date range");
+            }
+        } catch (e) {
+            console.error(e);
+            toast(`There was a problem downloading your usage data: ${e.message}`);
+        }
+
+        setIsSaving(false);
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" loading={isSaving} onClick={handleDownload}>
```

Comment:
Yah, I tried 🅱️ and it was feeling pretty cramped, so that's how I ended up with it where it is now. I agree it's kinda odd by itself there. I think we'll have the updated design ready shortly after this, so that was my rationale for having it on the right in it's own row initially. It's easy to move to either of those other spots though.